doc/api/n-api.md

```diff
@@ -3721,6 +3743,106 @@ NAPI_EXTERN napi_status napi_get_uv_event_loop(napi_env env,
+parameter whether the data was successfully added to the queue. Otherwise, the
+API blocks until space becomes available on the queue.
+`napi_call_threadsafe_function()` never blocks if the thread-safe function was
+created with a maximum queue size of 0.
+
+The callback given via the `call_js_cb` parameter is invoked on the main thread
+for each such call. If such a callback is not given, a default callback will be
+used, and the resulting JavaScript call will have no arguments. The `call_js_cb`
+callback receives the JavaScript function to call as a `napi_value` in its
+parameters, as well as the `void*` hint pointer used when creating the
+`napi_threadsafe_function`, and the next data pointer that was created by one of
+the secondary threads. The callback can then use an API such as
+`napi_call_function()` to call into JavaScript.
+
+*Note*: It is not necessary to call into JavaScript via `napi_make_callback()`
+because N-API runs `call_js_cb` in a context appropriate for callbacks.
+
+`napi_threadsafe_function` objects can be destroyed by calling
+`napi_delete_threadsafe_function()`. Make sure that all threads that have
+references to the `napi_threadsafe_function` object are stopped before deleting
+the object.
+
+### napi_create_threadsafe_function
+
+<!-- YAML
+added: REPLACEME
+-->
+```C
+NAPI_EXTERN napi_status
+napi_create_threadsafe_function(napi_env env,
+                                napi_value func,
+                                void* hint,
+                                size_t max_queue_size,
+                                napi_threadsafe_function_call_js call_js_cb,
+                                napi_threadsafe_function* result);
+```
+
+- `[in] env`: The environment that the API is invoked under.
+- `[in] func`: The JavaScript function to call from another thread.
+- `[in] hint`: Optional data to attach to the resulting `napi_threadsafe_function`.
+- `[in] max_queue_size`: Maximum size of the queue. 0 for no limit.
+- `[in] call_js_cb`: Optional callback which calls the JavaScript function in
+response to a call on a different thread. This callback will be called on the
+main thread. If not given, the JavaScript function will be called with no
+parameters and with the global object as its `this` value.
+- `[out] result`: The asynchronous thread-safe JavaScript function.
+
+### napi_call_threadsafe_function
+
+<!-- YAML
+added: REPLACEME
+-->
+```C
+NAPI_EXTERN napi_status
+napi_call_threadsafe_function(napi_threadsafe_function func,
+                              void* data,
+                              bool* result);
+```
+
+- `[in] func`: The asynchronous thread-safe JavaScript function to invoke. This
+API may be called from any thread.
+- `[in] data`: Data to send into JavaScript via the callback `call_js_cb`
+provided during the creation of the thread-safe JavaScript function.
+- `[out] result`: Optional flag receiving whether the data item was
```

Comment:
Personally I think this is a bit counter-intuitive. I would probably have a bool should_block_when_full, and return napi_error_queue_full when should_block_when_full= false. I don't feel strongly, though.